NEXUS-4893: Make Jetty use real logger

cstamas opened the pull request:
SLF4J and it's logback "backend" (is not backend but direct implementation)
moved into bundle. But, to avoid problems already described, logging
related packages are made "server classes", classes hidden from WAR,
as Nexus deliver the same stuff, and extends those by adding custom
appender implementations.

This change does/fixes:
- the original issue, as at "server level" it is not the SrdErrLogger trivial implementation used by Jetty to log (server logs and servlet logs) but logback (with Jetty's preferred native SLF4J support).
- the Jetty -- based on that item above -- now uses Slf4j and does not break the log "layout" (there was a similar issue, as pattern Jetty's logger used was different than Nexus pattern, making logs hard to read and parse)
- logging related classes are made "server classes" to not impose any requirements against Nexus WAR, hence, Nexus WAR wrt it's dependencies it ships in /lib folder are unchanged, it's still "fully functional" (and will not log to "noop" logger).

cstamas committed:
NEXUS-4893: Make Jetty use real logger

SLF4J and it's logback "backend" (is not backend but direct implementation)
moved into bundle. But, to avoid problems already described, logging
related packages are made "server classes", classes hidden from WAR,
as Nexus deliver the same stuff, and extends those by adding custom
appender implementations.

jdillon commented:
Cursory inspection looks good.  What does ci say?

velo commented:
+1
@jdillon I don't think this will affect CI since CI doesn't test the bundle itself.

@cstamas no changes need to nexus-bundles??

jdillon commented:
ci says this doesn't compile:

https://builds.sonatype.org/job/nexus-oss-feature/jdk=java-6x,label=linux/lastBuild/console

velo commented:
I could be wrong =D

cstamas committed:
NEXUS-4893: Fixing dependencies

Previous commit rendered build broken (compile errors). Mostly
due to wrongly set dependencies in nexus-logging-extras and
nexus-maven-plugin, as both of them declared logback-classic
with proper scope and both relied _implicitly_ on transitively
brought in logback-core, while both of these actually dependended
on core too (actual dependency not set in POM but implicitly relied
on their presence -- fragile builds). As Nexus parent pom got the
logback-base depMgt too (logback-classic was already present), these
modules got broken.

cstamas commented:
Started CI job:
https://builds.sonatype.org/view/nexus/job/nexus-oss-its-feature/285/

cstamas commented:
CI linux voted +1

peterlynch commented:
:beer:

cstamas commented:
Both slaves (lin and win) says +1
